### DISCUSSION TITLE
Complete selection without copy to Clipboard

### DISCUSSION THREAD

**Discussion by CarterLi** [wezterm/wezterm#4024] [Q&A]
I'd like to do this:

1. When left mouse button is up, complete the selection, but don't copy the selected text to clipboard.
2. When right mouse button is pressed, copy the selected to clipboard, or paste from clipboard if no text is selected.

I have done the second one with

```lua
config.mouse_bindings = {
    {
        event = { Down = { streak = 1, button = 'Right' } },
        action = wezterm.action_callback(function(window, pane)
            local has_selection = window:get_selection_text_for_pane(pane) ~= ''
            if has_selection then
                window:perform_action(act.CopyTo 'ClipboardAndPrimarySelection', pane)
                window:perform_action(act.ClearSelection, pane)
            else
                window:perform_action(act.PasteFrom 'Clipboard', pane)
            end
        end),
    }
}
```

However I found that currently `CompleteSelection` always triggers a copy. How can I avoid this behavior?

**Comment by vollink** [comment]
I would also be interested in such a feature.  Specifically, the difference between `primary `selection (linux normal select is available as middle click) vs `clipboard `selection (available from ctrl-shift-v), it seems every selection sets `clipboard`, which is unexpected.

**Reply by bew** [reply]
Feel free to change the default mouse bindings to only copy to the clipboard buffer you want
https://wezfurlong.org/wezterm/config/mouse.html#default-mouse-assignments

**Comment by bew** [comment]
> 1. When left mouse button is up, complete the selection, but don't copy the selected text to clipboard.

For this one, the `CompleteSelection` always does a copy indeed, but did you try do just remove the mouse up event, to avoid calling the `CompleteSelection` ?

**Comment by zw963** [comment]
Following code is working.

```lua
config.mouse_bindings = {
  {
    event = { Up = { streak = 1, button = 'Left' } },
    mods = 'NONE',
    action = act.Nop,
  },
```
